src/openforms/submissions/models.py

```diff
@@ -265,6 +272,41 @@ def get_merged_data_with_component_type(self) -> dict:
 
         return merged_data
 
+    def get_merged_appointment_data(self) -> dict:
+        merged_appointment_data = dict()
+        component_key_to_appointment_info = dict()
+        component_key_to_appointment_key = {
+            "showProducts": "productID",
+            "showLocations": "locationID",
+            "showTimes": "appStartTime",
+            "appointmentLastName": "clientLastName",
+            "appointmentBirthDate": "clientDateOfBirth",
+        }
+
+        for step in self.submissionstep_set.exclude(data=None).select_related(
+            "form_step"
+        ):
+            components = step.form_step.form_definition.configuration["components"]
+            flattened_components = get_flattened_components(components)
+            for component in flattened_components:
```

Comment:
doesn't it make more sense to use the `form.iter_components()` here instead, so you don't really have to concern yourself with the model relations?